File: problem-09/c/main.c
#include <stdio.h>
#include <stdbool.h>

bool is_pythagorean_triplet(int a, int b, int c) {
	return ((a * a) + (b * b)) == (c * c);
}

int solution(int n) {
	for (int a = 1; a < n; a++) {
		for (int b = 1; b < n; b++) {
			for (int c = 1; c < n; c++) {
				if (is_pythagorean_triplet(a, b, c) && ((a + b + c) == n)) {
					return a * b * c;
				}
			}
		}
	}

	return 0;
}

int main() {
	printf("Solution #9: %d\n", solution(1000));
}

File: problem-09/rust/src/main.rs
fn main() {
    println!("Solution #9: {}", solution(1000));
}

fn solution(n: i32) -> i32 {
    for a in 1..n {
        for b in 1..n {
            for c in 1..n {
                if is_pythagorean_triple(a,b,c) && ((a + b + c) == n) {
                    return a * b * c
                }
            }
        }
    }

    return 0
}

fn is_pythagorean_triple(a: i32, b: i32, c: i32) -> bool {
    return ((a * a) + (b * b)) == (c * c)
}
